specification/web/resource-manager/Microsoft.Web/stable/2019-08-01/StaticSites.json

```diff
@@ -0,0 +1,1703 @@
+          {
+            "$ref": "#/parameters/resourceGroupNameParameter"
+          },
+          {
+            "name": "name",
+            "in": "path",
+            "description": "Name of the static site.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "$ref": "#/parameters/subscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK.",
+            "schema": {
+              "$ref": "#/definitions/StaticSiteFunctionOverviewCollection"
+            }
+          },
+          "default": {
+            "description": "App Service error response.",
+            "schema": {
+              "$ref": "./CommonDefinitions.json#/definitions/DefaultErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Gets the functions of a static site": {
+            "$ref": "./examples/ListStaticSiteFunctions.json"
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Web/staticSites/{name}/resetapikey": {
+      "post": {
+        "tags": [
+          "StaticSites"
+        ],
+        "summary": "Resets the api key for an existing static site.",
+        "description": "Description for Resets the api key for an existing static site.",
+        "operationId": "StaticSites_ResetStaticSiteApiKey",
+        "parameters": [
+          {
+            "$ref": "#/parameters/resourceGroupNameParameter"
+          },
+          {
+            "name": "name",
+            "in": "path",
+            "description": "Name of the static site.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "resetPropertiesEnvelope",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/StaticSiteResetPropertiesARMResource"
+            }
+          },
+          {
+            "$ref": "#/parameters/subscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK."
+          },
+          "default": {
+            "description": "App Service error response.",
+            "schema": {
+              "$ref": "./CommonDefinitions.json#/definitions/DefaultErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Reset the api key for a static site": {
+            "$ref": "./examples/ResetStaticSiteApiKey.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Web/staticSites/{name}/secrets/list": {
```

Comment:
can you call this action as ListSecrets. The way its currently modeled, ARM will have a problem with the segments as there are resourceType/{name} segment. and action is over an instance. in current model, secrets becomes the resource type name and list becomes its instance, so there is no action .

Reply:
I see so the preference here is to use /staticSites/{name}/listSecrets?